QA Changes

dmytrotkk opened the pull request:
- add support of ima predeployed contracts calls and functions
- add support of paymaster contract calls and functions
- add support of config controller predeployed contract calls and functions
- add support of ima predeployed contracts calls and functions
- add support of etherbase predeployed contracts calls and functions
- add support of context predeployed contracts calls and functions
- add support of context predeployed contracts calls and functions

EvgeniyZZ committed:
Update README.md

test commit

EvgeniyZZ committed:
initial commit

EvgeniyZZ committed:
Update README.md

EvgeniyZZ committed:
upd

EvgeniyZZ committed:
upd

EvgeniyZZ committed:
Create README.md

EvgeniyZZ committed:
update gitignore

EvgeniyZZ committed:
add

EvgeniyZZ committed:
Update .gitignore

EvgeniyZZ committed:
update test

EvgeniyZZ committed:
add community locker

EvgeniyZZ committed:
add community pool

EvgeniyZZ committed:
speling fix

EvgeniyZZ committed:
add token manager erc721

EvgeniyZZ committed:
add token manager erc20

EvgeniyZZ committed:
update token managers

EvgeniyZZ committed:
fix fn naming

EvgeniyZZ committed:
refactor

EvgeniyZZ committed:
upd

EvgeniyZZ committed:
add contracts info

EvgeniyZZ committed:
update contracts structure

EvgeniyZZ committed:
Fix import

EvgeniyZZ committed:
addERC1155

EvgeniyZZ committed:
add community pool additional requests

EvgeniyZZ committed:
upd

EvgeniyZZ committed:
test

EvgeniyZZ committed:
fix contracts info

EvgeniyZZ committed:
update erc721

EvgeniyZZ committed:
add erc721withmetadate

EvgeniyZZ committed:
update imports

EvgeniyZZ committed:
update imoprt

EvgeniyZZ committed:
update erc1155

EvgeniyZZ committed:
update erc1155 token

EvgeniyZZ committed:
update erc721wmd

EvgeniyZZ committed:
add import

EvgeniyZZ committed:
update erc721wmd

EvgeniyZZ committed:
add call to map

EvgeniyZZ committed:
restructure, added tm_linker on schian

EvgeniyZZ committed:
update contracts info, dependencies, spelling fixes

EvgeniyZZ committed:
upd

EvgeniyZZ committed:
fix call

EvgeniyZZ committed:
upd

EvgeniyZZ committed:
update locker

EvgeniyZZ committed:
update ERC721

EvgeniyZZ committed:
add exit function for erc721

EvgeniyZZ committed:
Update token_manager_linker.py

EvgeniyZZ committed:
Update token_manager_erc721_wmt.py

EvgeniyZZ committed:
Update token_manager_erc721.py

EvgeniyZZ committed:
Update token_manager_erc1155.py

EvgeniyZZ committed:
Update deposit_box_erc721.py

EvgeniyZZ committed:
add direct deposit for erc1155

EvgeniyZZ committed:
tiny fix

EvgeniyZZ committed:
spelling fix

EvgeniyZZ committed:
remove unsused imports

EvgeniyZZ committed:
pep8

EvgeniyZZ committed:
remove unused imports

EvgeniyZZ committed:
fix E231

EvgeniyZZ committed:
update skale-contracts

EvgeniyZZ committed:
add paymaster

EvgeniyZZ committed:
fix naming

EvgeniyZZ committed:
update skale-contracts version

EvgeniyZZ committed:
add paymaster

EvgeniyZZ committed:
add paymaster

EvgeniyZZ committed:
update workflow test

EvgeniyZZ committed:
fix flake8 test paymaster

yatsunastya committed:
add new functions, fix bugs

yatsunastya committed:
added skipTime and effectiveTimestamp

yatsunastya committed:
Pull changes from develop

yatsunastya committed:
fixed flake8 issues

yatsunastya committed:
Merge pull request #2 from EvgeniyZZ/paymaster-fix

Paymaster edit

EvgeniyZZ committed:
update version

EvgeniyZZ committed:
Merge branch 'develop' of github.com:EvgeniyZZ/skale.py into develop

EvgeniyZZ committed:
rollback  global version

EvgeniyZZ committed:
add revoke role

EvgeniyZZ committed:
fix flake

yatsunastya committed:
added debug contracts option

yatsunastya committed:
Merge pull request #3 from EvgeniyZZ/paymaster-fix

added debug contracts option

yatsunastya committed:
added debts calls

yatsunastya committed:
Merge pull request #4 from EvgeniyZZ/paymaster-fix

Added debts calls

OleksanderSalamatov committed:
add full config controller support

OleksanderSalamatov committed:
add newline

OleksanderSalamatov committed:
fix linter errors

OleksanderSalamatov committed:
remove trailing whitespace

OleksanderSalamatov committed:
chang str output to bytes, change ContractFunction to TxRes

OleksanderSalamatov committed:
add etherbase predeployed full functionality

OleksanderSalamatov committed:
Fix linter warning

OleksanderSalamatov committed:
add context contract full functionality

OleksanderSalamatov committed:
fix linter errors

EvgeniyZZ committed:
fix contract fn

EvgeniyZZ committed:
Merge pull request #5 from EvgeniyZZ/add-config-controller

Add config controller

EvgeniyZZ committed:
Merge pull request #7 from EvgeniyZZ/add-context

Add context contract

EvgeniyZZ committed:
Merge branch 'develop' into add-etherbase

EvgeniyZZ committed:
Merge pull request #6 from EvgeniyZZ/add-etherbase

Add etherbase

EvgeniyZZ committed:
clean up gitignore

EvgeniyZZ committed:
restore workflow

EvgeniyZZ committed:
fix

EvgeniyZZ committed:
fix

OleksanderSalamatov committed:
fix typo and formatting

EvgeniyZZ committed:
Merge pull request #8 from EvgeniyZZ/fix-typo

fix typo and formatting

EvgeniyZZ committed:
fix type

EvgeniyZZ committed:
Merge branch 'develop' into develop

EvgeniyZZ committed:
add license

dmytrotkk committed:
Merge pull request #566 from EvgeniyZZ/develop

Add IMA perdeployed

dmytrotkk committed:
Fix linter in config controller

dmytrotkk committed:
Update docker compose command

oleksandrSydorenkoJ committed:
qa-changes: add transfer_to_schain_hash_erc20_direct function

oleksandrSydorenkoJ committed:
qa-changes: add transferToSchainERC20Direct function

oleksandrSydorenkoJ committed:
qa-changes: Fix transferToSchainERC20Direct function

OleksanderSalamatov committed:
add strict version of eth_typing to setup.py

OleksanderSalamatov committed:
fix comma

OleksanderSalamatov committed:
update eth_typing version to match qa-automation repo req-s

dmytrotkk committed:
Merge latest changes from develop

dmytrotkk committed:
Merge latest changes from develop

badrogger commented on skale/contracts/ima/mainnet/community_pool.py:
```diff
@@ -0,0 +1,72 @@
+        schain_id = keccak_hash.digest()
+        return self.contract.functions.checkUserBalance(schain_id, receiver).call()
+
+    def get_recommended_recharge_amount(self, schain_hash: bytes, receiver: int) -> int:
```
should be schain_name: str instead.

badrogger commented on skale/contracts/ima/schain/community_locker.py:
```diff
@@ -0,0 +1,71 @@
+    def last_message_timestamp(self, address) -> int:
+        return self.contract.functions.lastMessageTimeStamp(address).call()
+
+    def time_limit_per_msg(self, schain_hash: bytes) -> int:
```
should be schain_name: str

badrogger commented on skale/contracts/ima/schain/token_manager_erc1155.py:
```diff
@@ -0,0 +1,96 @@
+            schain_name, token_address, token_ids, amount
+        )
+
+    def get_clones_erc1155(self, schain_hash: bytes, address: str) -> int:
```
should be schain_name: str

badrogger commented on skale/contracts/ima/schain/token_manager_erc20.py:
```diff
@@ -0,0 +1,93 @@
+    def get_role_member(self, role: bytes, index: int) -> bytes:
+        return self.contract.functions.getRoleMember(role, index).call()
+
+    def get_clones_erc20(self, schain_hash: bytes, address: str) -> int:
```
should be schain_name: str

badrogger commented on skale/contracts/ima/schain/token_manager_erc721_wmt.py:
```diff
@@ -0,0 +1,77 @@
+    def get_role_member(self, role: bytes, index: int) -> bytes:
+        return self.contract.functions.getRoleMember(role, index).call()
+
+    def get_clones_erc721(self, schain_hash: bytes, address: str) -> int:
```
should be schain_name: str

Masterix0 committed:
Update typing and variable names:
- from `schain_hash: bytes` to `schain_name: str`;
- from `address: str` to `address: ChecksumAddress`.

Masterix0 committed:
Update test runner to 22.04

Masterix0 committed:
Update test_allocator runner to 22.04

Masterix0 committed:
Changed `schain_name: str` to `schain_name: SchainName` in whole codebase

dmytrotkk committed:
Merge branch 'develop' into qa-changes
